samples/seldon/mnist_tf_nopush.py

```diff
@@ -0,0 +1,180 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from kubernetes import client as k8s_client
+import kfp.dsl as dsl
+import json
+from string import Template
+
+@dsl.pipeline(
+    name="Seldon MNIST TF",
+    description="Example of training and serving seldon MNIST TF model. Like kubeflow/example-seldon but using existing images."
+)
+
+def mnist_tf(docker_repo_training='seldonio/deepmnistclassifier_trainer',
+             docker_tag_training='0.3',
+             docker_repo_serving='seldonio/deepmnistclassifier_runtime',
+             docker_tag_serving='0.3'):
+
+#use volume for storing model
+#here model is saved and mounted into pre-defined image for serving
+#alternatively model can be baked into image - for that see mabdeploy-seldon.py
+#requires seldon v0.3.0 or higher
+    modelvolop = dsl.VolumeOp(
+        name="modelpvc",
+        resource_name="modelpvc",
+        size="50Mi",
+        modes=dsl.VOLUME_MODE_RWO
+    )
+
+    tfjobjsonTemplate = Template("""
+{
+	"apiVersion": "kubeflow.org/v1beta1",
+	"kind": "TFJob",
+	"metadata": {
+		"name": "mnist-train-{{workflow.uid}}",
+		"ownerReferences": [
+		{
+			"apiVersion": "argoproj.io/v1alpha1",
+			"kind": "Workflow",
+			"controller": true,
+			"name": "{{workflow.name}}",
```

Comment:
Is there any way to avoid the `{{workflow.name}}` placeholder? The `{{workflow.uid}}` placeholder will soon be available as `dsl.run_id_placeholder`.

Reply:
Note that this use is just an ownerReference determining that if the argo job resource is cleaned up then the TFJob is cleaned up with it as a matter of garbage collection. It's not being used to set unique names - the workflow.uid is doing that. I'm confused as to workflow.name could be something to avoid in this case.